### PR TITLE
feat(autoware_lidar _ransfusion): fix 3d bounding box orientation

### PR DESCRIPTION
**Pull request by SamratThapa120**
## Description
We found that the 3d bounding boxes were rotated by 90 degrees for transfusion
image

This PR fixes the issue.

### CODE REVIEW

File: perception/autoware_lidar_transfusion/lib/ros_utils.cpp [hunk 1]
```diff
@@ -49,14 +49,14 @@ void box3DToDetectedObject(
 
   // pose and shape
   // mmdet3d yaw format to ros yaw format
-  float yaw = box3d.yaw + autoware::universe_utils::pi / 2;
+  float yaw = box3d.yaw;
   obj.kinematics.pose_with_covariance.pose.position =
```
Comment (amadeuszsz) [inline perception/autoware_lidar_transfusion/lib/ros_utils.cpp:53]: If we don't need any format conversion, please remove this line and upper comment as well (line 51). We can use `box3d.yaw` directly.

File: perception/autoware_lidar_transfusion/lib/ros_utils.cpp [hunk 2]
```diff
@@ -49,14 +49,14 @@ void box3DToDetectedObject(
...
     autoware::universe_utils::createPoint(box3d.x, box3d.y, box3d.z);
   obj.kinematics.pose_with_covariance.pose.orientation =
     autoware::universe_utils::createQuaternionFromYaw(yaw);
   obj.shape.type = autoware_perception_msgs::msg::Shape::BOUNDING_BOX;
   obj.shape.dimensions =
-    autoware::universe_utils::createTranslation(box3d.length, box3d.width, box3d.height);
+    autoware::universe_utils::createTranslation(box3d.width, box3d.length, box3d.height);
```
Comment (amadeuszsz) [inline perception/autoware_lidar_transfusion/lib/ros_utils.cpp:59]: Changing order does not match with `createTranslation` funciton call (x, y, z). If we need to do a switch here, it should be done in [kernel](https://github.com/autowarefoundation/autoware.universe/blob/c2438fd2841bce7792be4bc57e88edbe6b34276d/perception/autoware_lidar_transfusion/lib/postprocess/postprocess_kernel.cu#L86-L87). However, the operation in kernel match indices in forward propagation ([TIER IV internal link](https://github.com/tier4/autoware-ml/blob/bf604baf6317d76deb08e78a5327615e8f27ccd0/projects/TransFusion/transfusion/transfusion_head.py#L683-L684)). Could you check training pipeline if there is a mistake?

**Review by amadeuszsz — CHANGES_REQUESTED** [review]
Nice catch! I shared my comments

File: perception/autoware_lidar_transfusion/lib/ros_utils.cpp (hunk 1 above)
Comment (SamratThapa120) [reply perception/autoware_lidar_transfusion/lib/ros_utils.cpp:51]: FIXED

**Review by amadeuszsz — APPROVED** [review]
LGTM. Thanks for fix! 🚀

**Comment by kminoda** [comment]
@SamratThapa120 Would you follow the PR format as others do? Also, if you've run the evalation on Evaluator, please post the link as well.

**Comment by SamratThapa120** [comment]
@SamratThapa120 Thank you for the information. I updated the PR description. I havent yet run the evaluation on Evaluator, I will post the link once it is done.

**Pull request merged**